core/client/fs/src/main/java/alluxio/client/file/FileSystemUtils.java

```diff
@@ -153,8 +153,10 @@ public static void persistAndWait(final FileSystem fs, final AlluxioURI uri,
   public static void persistAndWait(final FileSystem fs, final AlluxioURI uri,
       long persistenceWaitTime, int timeoutMs) throws FileDoesNotExistException, IOException,
       AlluxioException, TimeoutException, InterruptedException {
-    fs.persist(uri, ScheduleAsyncPersistencePOptions
-        .newBuilder().setPersistenceWaitTime(persistenceWaitTime).build());
+    if (!fs.getStatus(uri).getPersistenceState().equals("TO_BE_PERSISTED")) {
```

Comment:
Should this check be done on the server side, since this will now require 2 RPCs for each persist request. Instead, maybe the server side should perform this check?

Reply:
Thanks @gpang that make sense